ee/modules/030-cloud-provider-openstack/hooks/handle_discovery_data_volume_types.go

```diff
@@ -6,67 +6,37 @@ Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https
 package hooks
 
 import (
-	"os"
 	"regexp"
 	"sort"
 	"strings"
 	"unicode"
 
 	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
-	"github.com/flant/addon-operator/sdk"
-)
-
-var _ = sdk.RegisterFunc(&go_hook.HookConfig{
-	OnBeforeHelm: &go_hook.OrderedConfig{Order: 20},
-	Schedule: []go_hook.ScheduleConfig{
-		{
-			Name:    "discover_volume_types",
-			Crontab: "45 * * * *",
-		},
-	},
-}, handleDiscoverVolumeTypes)
-
-type storageClass struct {
-	Name string `json:"name"`
-	Type string `json:"type"`
-}
 
-func handleDiscoverVolumeTypes(input *go_hook.HookInput) error {
-	err := initOpenstackEnvs(input)
-	if err != nil {
-		return err
-	}
-
-	var openstackVolumeTypes []string
-	if os.Getenv("D8_IS_TESTS_ENVIRONMENT") != "" {
-		openstackVolumeTypes = []string{"__DEFAULT__", "some-foo", "bar", "other-bar", "SSD R1", "-Xx__$()? -foo-", "  YY fast SSD-foo."}
-	} else {
-		openstackVolumeTypes, err = getVolumeTypesArray()
-		if err != nil {
-			return err
-		}
-	}
+	"github.com/deckhouse/deckhouse/go_lib/cloud-data/apis/v1alpha1"
+)
 
-	storageClassesMap := make(map[string]string, len(openstackVolumeTypes))
+func handleDiscoveryDataVolumeTypes(input *go_hook.HookInput, volumeTypes []v1alpha1.OpenStackCloudProviderDiscoveryDataVolumeType) {
```

Comment:
We use this function in one place in discoverer hook. Let move this function in the hook file.